app/views/layouts/application.html.erb

```diff
@@ -13,9 +13,27 @@
       <%= link_to "Home Page", "/home" %>
       <%= link_to "All Merchants", "/merchants" %>
       <%= link_to "All Items", "/items" %>
-      <%= link_to "Cart: #{cart.total_items}", "/cart" %>
-      <%= link_to "Register New User", "/register" %>
-      <%= link_to "User Log In", "/login" %>
+      <% if current_merchant? %>
+        <%= link_to "Cart: #{cart.total_items}", "/cart" %>
+        <%= link_to "Merchant's Dashboard", "/merchant" %>
+        <%= link_to "Logout", "/logout", method: :delete %>
+        <%= content_tag :p, "Logged in as #{current_user.name}" %>
+      <% elsif current_admin? %>
+        <%= link_to "Admin's Dashboard", "/admin" %>
+        <%= link_to "All Users", "/admin/users" %>
+        <%= link_to "Logout", "/logout", method: :delete %>
+        <%= content_tag :p, "Logged in as #{current_user.name}" %>
+      <% elsif current_user %>
+        <%= link_to "Cart: #{cart.total_items}", "/cart" %>
+        <%= link_to "#{current_user.name}'s Profile", "/profile" %>
+        <%= link_to "Logout", "/logout", method: :delete %>
+        <%= content_tag :p, "Logged in as #{current_user.name}" %>
+      <% else %>
+        <%= link_to "Cart: #{cart.total_items}", "/cart" %>
+        <%= link_to "Register New User", "/register" %>
+        <%= link_to "User Log In", "/login" %>
+      <% end %>
```

Comment:
Looks great! Maybe we can refactor these into partials in the future...

Reply:
That's a great idea!